### DISCUSSION TITLE
PDF NOT LOADING LATEST VERSION

### DISCUSSION THREAD

**Discussion by aathi271991** [stephanrauh/ngx-extended-pdf-viewer#2499] [General]
I am updating from version 20.5.2 to 21.1.2, but the PDF is not loading and I am getting the error:

ERROR:Uncaught TypeError: Promise.withResolvers is not a function
    at viewer-4.5.760.min.mjs:46:267624
    
      _initializedCapability: {
        ...Promise.withResolvers(),
        settled: !1
    },
In the previous version, it worked correctly. What should I do to resolve this issue?
![Screenshot 2024-08-13 215417](https://github.com/user-attachments/assets/b87824ad-c741-4339-bca5-c077d2114a0b)

**Comment by stephanrauh** [comment]
Which browser are you using? And are you using the latest version of zone.js?

Outdated versions of zone.js are known to cause problems. 

If you're using the latest version of zone.js and the problem still persists, please debug the code and tell me what happens. I'd do it myself, but the error doesn't show on my machine. Here's what to do:

- add the attribute `[useInlineScripts]="false"`. This forces the viewer to load the browser switch from an external file. That's less efficient, but it makes debugging easier.
- Run your application.
- Open the file "op-chaining-support.js" in the "sources" tab of your browser's developer tool and add a breakpoint, as shown in the image below.
- Reload the page to trigger the browser switch again.
- If your browser doesn't support modern JavaScript (like `Promise.withResolvers`), the method `supportsPromiseWithResolvers()` returns false. (That's slightly confusing - the variable has the wrong name, but the method works nonetheless). Also, `window.ngxExtendedPdfViewerCanRunModernJSCode` should become false.
- If so, the viewer loads the JavaScript files called "*-es5.mjs". 

Can you confirm this?

Update: I just noticed that the *-es5.mjs files use `Promise.withResolvers()`. That's clearly a bug. Nonetheless, it'd be nice if you went through the steps above and tell me what happens on your machine. Otherwise, it's all guesswork. I hope you'll manage in less then ten minutes.

<img width="763" alt="image" src="https://github.com/user-attachments/assets/6d42c4a0-075d-4788-91fb-339d71980084">

**Comment by garciajavier** [comment]
I have the same problem. I did what you say and I get window.ngxExtendedPdfViewerCanRunModernJSCode = 'true'

**Reply by stephanrauh** [reply]
If you're talking about the problem mentioned [here](https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/2362#issuecomment-2659115701), the root problem is the JavaScript files aren't there. From the outside, the error looks identical, but that's a different cause.
